### PR TITLE
Fixed a bug in the stories

### PR DESCRIPTION
**Pull request by kiddinn**
This fixes at least partially #2224 

This fixes issues that came up during the transition to open search, at least in the notebook.

There is also a fix for saved searches inside the API client, where the indices were not correctly restored/read from the backend.

Before one could not get the markdown content of a story... now...

image

as you can see, a markdown version of the story can be produced inside a notebook.

**Checks**

- [x] All tests succeed.
- [x] Unit tests added.
- [x] e2e tests added.
- [ ] Documentation updated.

### CODE REVIEW

File: api_client/python/timesketch_api_client/search.py [hunk 1]
```diff
@@ -249,6 +249,11 @@ def add_end_time(self, end_time):
         """
         if end_time.endswith("Z"):
             end_time = end_time[:-1]
+
+        # Check for a whole day, YYYY-MM-DD.
+        if len(end_time) == 10:
```
Comment (berggren) [inline api_client/python/timesketch_api_client/search.py:254]: Do we want to check for the format here as well, and not just the length of the string?
`YYYY-MM-DD`

File: timesketch/lib/stories/api_fetcher.py [hunk 2]
```diff
@@ -165,6 +173,31 @@ def get_aggregation_group(self, agg_dict):
...
+            A tuple with two items, a list of indices and a list of
+            timeline IDs.
+        """
+        indices = []
+        timeline_ids = []
+
+        if isinstance(index_list, str):
+            index_list = index_list.split(',')
+
+        for item in index_list:
```
Comment (berggren) [inline timesketch/lib/stories/api_fetcher.py:193]: `item` is a bit generic here, can you rename to `index` ?

**Review by berggren — CHANGES_REQUESTED** [review]
A coupe of things, I know this is in draft already.

File: timesketch/lib/stories/api_fetcher.py (hunk 2 above)
Comment (kiddinn) [reply timesketch/lib/stories/api_fetcher.py:193]: yes, changed it

File: api_client/python/timesketch_api_client/search.py (hunk 1 above)
Comment (kiddinn) [reply api_client/python/timesketch_api_client/search.py:254]: using re

**Review by kiddinn — COMMENTED** [review]
made some changes

**Pull request merged**